chef/cookbooks/provisioner/templates/default/crowbar_join.suse.sh.erb

```diff
@@ -301,29 +301,6 @@ do_chef_client() {
         fi
     fi
 
-    # we still didn't succeed with chef-client, so let's try running it again
-    # in "installed" state, to pretend we just come straight from a fresh
-    # install
-    echo_debug "Failed to run chef-client, trying with state \"installed\""
-    post_state $HOSTNAME "installed"
-
-    echo_debug "Syncing Time"
-    sync_time
-    echo_debug "Removing Chef Cache"
-    rm -rf /var/cache/chef/*
-    echo_debug "Checking Keys"
-    rm -f /etc/chef/client.pem
-
-    echo_verbose "Running Chef Client (try 3, pass 1) - password cleanup"
-    if log_to chef chef-client $CHEF_CLIENT_OPTIONS; then
-	# it worked, cool, let's try again with "readying" state
-        post_state $HOSTNAME "readying"
-        echo_verbose "Running Chef Client (try 3, pass 2) - password cleanup"
-        if log_to chef chef-client $CHEF_CLIENT_OPTIONS; then
-            return
-        fi
-    fi
-
```

Comment:
I would also kill the cache cleanup that is a few lines above; it makes no sense at all, imho. Let's just move to recovering state, run chef (to validate that chef still works), and if successful run chef in readying again.